tools/git-prompt.sh

```diff
@@ -255,7 +255,7 @@ __git_ps1_colorize_gitstring ()
 	local flags_color="$c_lblue"
 
 	local branch_color=""
```

Comment:
Actually, this file `tools/git-prompt.sh` is just a copy of the upstream [git/git `contrib/completion/git-prompt.sh`](https://github.com/git/git/blob/master/contrib/completion/git-prompt.sh), so I would like to avoid edit this file independently from the upstream version.

Hmm, these seem not to have yet been fixed in the upstream either.

- https://github.com/git/git/blob/0607f793cbe0af16aee6d2480056d891835884bd/contrib/completion/git-prompt.sh#L272
- https://github.com/git/git/blob/0607f793cbe0af16aee6d2480056d891835884bd/contrib/completion/git-prompt.sh#L301
- https://github.com/git/git/blob/0607f793cbe0af16aee6d2480056d891835884bd/contrib/completion/git-prompt.sh#L499

Reply:
Gotchia :+1: Yeah it would be better to update upstream

Reply:
I have created patches for the upstream Git.

https://github.com/akinomyoga/git/commit/ce955e4f0293214aa8171a789bec9f073f7913e2
https://github.com/akinomyoga/git/commit/a7799fb5a3d65cb48db345e37c8f87a62e2c7fba

I haven't included the change `(* | default)` -> `(default | *)` in the case statement.

To submit patches to git, we need to include the `Signed-off-by` line for the contributed people in the commit message.

```
Signed-off-by: Edwin Kofler <edwin@kofler.dev>
```

@hyperupcall Can you agree on submitting the patches to the upstream git and adding the above line in the commit messages of the patches? The comments to the patches are also welcome.

Reply:
Yes! I will submit these patches to upstream Git and add the `Signed-off-by: Edwin Kofler <edwin@kofler.dev>` line

The `*|default` thing only fixes a ShellCheck issue, so I don't see any strong reason to include it in the patches

Reply:
Thank you for your reply! I have updated the patches.

https://github.com/akinomyoga/git/commit/e80559019501ba8f4b942e92257fea84baa69f8c
https://github.com/akinomyoga/git/commit/2da42103fbe55aed548a543d1bf4aeace8179b71

Ah, I intended to submit the patches to `git@vger.kernel.org` but will you submit the patches?

Anyway, I first submit them to myself (with Cc to you) for testing purposes.

Reply:
Ohh, you meant that you were going to submit the patches, I see what you mean now :sweat_smile: 

It might be better for you to submit them since it seems you were already going to do so? I don't mind either way, but it seems more right to go with what you originally planned

Reply:
Also I got your emails, I'll check the patch contents

Reply:
The patches look good to me, and I applied them fine

Reply:
Thanks!

Reply:
I have submitted them.